pkg/store/event/event.go

```diff
@@ -44,15 +44,135 @@ type eventStore struct {
 	db *dbstore.DB
 }
 
-// TODO: implement core.EventStore interface
-func (e *eventStore) List(context.Context) ([]*core.Event, error) { return nil, nil }
-func (e *eventStore) ListByExperiment(context.Context, string, string) ([]*core.Event, error) {
-	return nil, nil
+// FindPodRecordsByEventID return the list of PodRecords according to the eventID
+func (e *eventStore) FindPodRecordsByEventID(_ context.Context, id uint) ([]*core.PodRecord, error) {
+	pods := make([]*core.PodRecord,0)
+	if err := e.db.Where(
+		"event_id = ?", id).
+		Find(&pods).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+		return nil, err
+	}
+	return pods, nil
+}
+
+// List returns the list of events
+func (e *eventStore) List(_ context.Context) ([]*core.Event, error) {
+	var resList []core.Event
+	eventList := make([]*core.Event, 0)
+
+	if err := e.db.Find(&resList).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+		return nil, err
+	}
+
+	for _, et := range(resList) {
+		pods, err:= e.FindPodRecordsByEventID(context.Background(), et.ID)
+		if err != nil {
+			return nil, err
+		}
+		et.Pods = pods
+		eventList = append(eventList, &et)
+	}
+
+	return eventList, nil
+}
+
+// ListByExperiment returns a event list by the name and namespace of the experiment.
+func (e *eventStore) ListByExperiment(_ context.Context, namespace string, experiment string) ([]*core.Event, error) {
+	var resList []core.Event
+
+	if err := e.db.Where(
+		"namespace = ? and experiment = ? ",
+		namespace, experiment).
+		Find(resList).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+		return nil, err
+	}
+
+	eventList := make([]*core.Event, 0, len(resList))
+	for _, et := range resList {
+		pods, err:= e.FindPodRecordsByEventID(context.Background(), et.ID)
+		if err != nil {
+			return nil, err
+		}
+		et.Pods = pods
+		eventList = append(eventList, &et)
+	}
+
+	return eventList, nil
 }
-func (e *eventStore) ListByPod(context.Context, string, string) ([]*core.Event, error) {
-	return nil, nil
+
+// ListByNamespace returns the list of events according to the namespace
+func (e *eventStore) ListByNamespace(_ context.Context, namespace string) ([]*core.Event, error) {
```

Comment:
why we leave context _?

Reply:
This is a common pattern if this parameter is not used.

Reply:
Because everything needed is in the DB in the eventStore, the context is temporarily not used.  I don't know if the context will be used in subsequent changes.